src/Controller/RestaurantController.php

```diff
@@ -214,52 +236,71 @@ public function listAction(Request $request,
             ]);
         }
 
-        $page = $request->query->getInt('page', 1);
-        $offset = ($page - 1) * self::ITEMS_PER_PAGE;
+        // find cuisines which can be selected by user to filter
+        $cuisines = $repository->findExistingCuisines();
 
-        if ($request->query->has('geohash') && strlen($request->query->get('geohash')) > 0) {
-            $geotools = new Geotools();
-            $geohash = $request->query->get('geohash');
+        $cacheKey = $this->getShopsListCacheKey($request, $type);
 
-            $decoded = $geotools->geohash()->decode($geohash);
+        // for filtering we need in query param the full type instead of the key
+        $request->query->set('type', $type);
```

Comment:
Why do we need to always have a `type`? The following URL is valid `/shops?category=exclusive`, and should return all exclusive shops, no matter the type.